### ISSUE TITLE
[Bug] parameter "max_sequence_length" is not added into the mlc-chat-config.json while compiling model from HuggingFace

### ISSUE THREAD

**Issue by JackZeng0208** [mlc-ai/mlc-llm#819]
## 🐛 Bug

When I follow the instruction [Compile Model via MLC](https://mlc.ai/mlc-llm/docs/compilation/compile_models.html) to compile Llama-2-7B-chat model from HuggingFace,  it doesn't have "max_sequence_length" parameter in mlc-chat-config.json

## To Reproduce
Run command:
```
python -m mlc_llm.build --hf-path meta-llama/Llama-2-7b-chat-hf --target android --max-seq-len 768 --quantization q4f16_1
```
## Error Message
```
Finish exporting chat config to dist\Llama-2-7b-chat-hf-q4f16_1\params\mlc-chat-config.json
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "D:\mlc-llm\mlc_llm\build.py", line 13, in <module>
    main()
  File "D:\mlc-llm\mlc_llm\build.py", line 10, in main
    core.build_model_from_args(parsed_args)
  File "D:\mlc-llm\mlc_llm\core.py", line 578, in build_model_from_args
    mod = mod_transform_before_build(mod, param_manager, args, config)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\mlc-llm\mlc_llm\core.py", line 361, in mod_transform_before_build
    max_seq_len = config["max_sequence_length"]
                  ~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'max_sequence_length'
```
mlc-chat-config.json:
```json
{
    "model_lib": "Llama-2-7b-chat-hf-q4f16_1",
    "local_id": "Llama-2-7b-chat-hf-q4f16_1",
    "conv_template": "llama-2",
    "temperature": 0.7,
    "repetition_penalty": 1.0,
    "top_p": 0.95,
    "mean_gen_len": 128,
    "max_gen_len": 512,
    "shift_fill_factor": 0.3,
    "tokenizer_files": [
        "tokenizer.json",
        "tokenizer.model"
    ],
    "model_category": "llama",
    "model_name": "Llama-2-7b-chat-hf"
}
```

## Potential Reason:
In `dump_mlc_chat_config()` under core.py, the function doesn't dump "max_sequence_length" parameter into JSON file.

**Comment by junrushao** [comment]
CC: @nverke @masahi @Hzfengsy

**Comment by masahi** [comment]
Should be fixed by https://github.com/mlc-ai/mlc-llm/pull/806, can somebody merge this?

**Comment by MasterJH5574** [comment]
Just merged #806. Would you like to confirm again?

**Comment by junrushao** [comment]
Thanks @MasterJH5574 @masahi for quickly acting on this!
